shoop/core/models/_orders.py

```diff
@@ -684,5 +684,42 @@ def can_edit(self):
 OrderLogEntry = define_log_model(Order)
 
 
+# note - @pikkupomo mentioned not to use a Polymorphic model due to migration troubles; hopefully multi-table
+# inheritance is better (seems to just be syntactic sugar around a one-to-one field with the parent)
+# One problem with defining the Purchase Order this way is upon PurchaseOrder creation - see _creator.py
+# another bummer is having to do Order.objects.exclude(purchaseorder__isnull=True) in order to get sales orders
+# although I suppose the latter can be fixed by adding a method to the query manager
+class PurchaseOrder(Order):
```

Comment:
Do we need them to have a concrete base class at all? What if we just move most of the stuff from Order to a new class which could be named e.g. `AbstractOrder`. Then just derive both `Order` and `PurchaseOrder` from the `AbstractOrder`. (Naming could also be `Order`/`SalesOrder`/`PurchaseOrder`, but that would force us to rename `Order` to `SalesOrder` everywhere, IMHO we shouldn't do that.)

Using polymorphic base class for a model is useful when you have a collection of objects which have the same interface, but possibly different implementation and you want to use them through the common interface. I don't think that's the case here.